packages/core/src/types/config.ts

```diff
@@ -1,6 +1,7 @@
-export type TaskConfig = 'on' | 'off' | ['on' | 'off', unknown];
+export type ActionConfig = { [key: string]: number | 'off' };
+export type TaskConfig = { actions?: ActionConfig[]; [key: string]: any } | [];
```

Comment:
Based on the above change, I think `ActionConfig` should be an object of key (action name) and value ('on', 'off', or object that defines a property `threshold`, and a number value)

```suggestion
export type TaskConfig = { actions?: ActionConfig; [key: string]: any } | [];
```

Reply:
Correct, but there can be multiple Actions configured per Task, so this still needs to be an array

Reply:
I was thinking `ActionConfig` would be an object like so:

```
"actions": {
  "numFoos": { threshold: 5 },
  "bars": "off"
}
```

Why does it need to be an array when it can be an object keyed off the action name?